src/zkevm_specs/evm_circuit/execution/error_oog_memory_copy.py

```diff
@@ -0,0 +1,66 @@
+from zkevm_specs.evm_circuit.table import CallContextFieldTag
+from zkevm_specs.util.param import (
+    GAS_COST_ACCOUNT_COLD_ACCESS,
+    GAS_COST_FASTEST,
+    GAS_COST_WARM_ACCESS,
+    N_BYTES_MEMORY_ADDRESS,
+)
+from ...util import FQ
+from ..instruction import Instruction
+from ..opcode import Opcode
+from ...util import N_BYTES_GAS
+
+
+def error_oog_memory_copy(instruction: Instruction):
+    # retrieve op code associated to oog constant error
+    opcode = instruction.opcode_lookup(True)
+    (
+        is_calldata_copy,
+        is_code_copy,
+        is_ext_code_copy,
+        is_return_data_copy,
+    ) = instruction.multiple_select(
+        opcode, (Opcode.CALLDATACOPY, Opcode.CODECOPY, Opcode.EXTCODECOPY, Opcode.RETURNDATACOPY)
+    )
+
+    # Constrain opcode must be one of CALLDATACOPY, CODECOPY, EXTCODECOPY or RETURNDATACOPY.
+    instruction.constrain_equal(
+        is_calldata_copy + is_code_copy + is_ext_code_copy + is_return_data_copy, FQ(1)
+    )
+
+    # EXTCODECOPY has one more extra stack pop for external address
+    if is_ext_code_copy == FQ(1):
+        external_address = instruction.stack_pop()
+    memory_offset_word = instruction.stack_pop()
```

Comment:
Do `memory_offset_word`, `calldata offset` and `copy_size_word` corresponds to `destination_offset`, `source_offset `and `copy_byte_size `as mentioned in the md file?

Reply:
It seems to be the case for the 1st and 3rd.

Reply:
I followd the namings of CODECOPY.md, CALLDATACOPY.md EXTCODECOPY.md and RETURNDATACOPY.md. `source_offset` isn't clear for me bcs it doesn't say which source. It could be either calldata or memory. I was trying to complete python code and didn't think about the refine the md file. Do you think it's better to make the name consistent (revise the var names in md file)?

Reply:
I followd the namings of CODECOPY.md, CALLDATACOPY.md EXTCODECOPY.md and RETURNDATACOPY.md. source_offset isn't clear for me bcs it doesn't say which source. It could be either calldata or memory. I was trying to complete python code and didn't think about the refine the md file. Do you think it's better to make the name consistent (revise the var names in md file)?

Reply:
I think it's better to make our wording consistent, fixed in [51a3e47](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/468/commits/51a3e47122fda33a4fa42e1f7be5e01d4fd6cef5)